include/mc_rtc/loader.hpp

```diff
@@ -157,6 +176,27 @@ T * ObjectLoader<T>::create(const std::string & name, Args... args)
   return ptr;
 }
 
+template<typename T>
+template<typename... Args>
+T * ObjectLoader<T>::create_from_callbacks(const std::string & name, Args... args)
+{
+  return callbacks_.call<T *, const typename std::decay<Args>::type &...>(name, std::forward<Args>(args)...);
+}
+
+template<typename T>
+template<typename RetT, typename... Args>
+void ObjectLoader<T>::register_object(const std::string & name, std::function<RetT *(const Args &...)> callback)
+{
+  if(has_object(name))
+  {
+    throw LoaderException(fmt::format("{} is already registered with this loader", name));
+  }
+  static_assert(std::is_base_of<T, RetT>::value,
+                "You cannot register an object that does not derive from the loader base-class");
```

Comment:
```suggestion
  static_assert(std::is_base_of<T, RetT>::value,
                "Object \"{}\" cannot be registered as it does not derive from the loader base-class", name);
```

Reply:
This is a static assertion so it cannot provide this information (but hopefully the compilation error will have the name somewhere :)